### DISCUSSION TITLE
How to export model.nemo from ckpt?

### DISCUSSION THREAD

**Discussion by Minxiangliu** [NVIDIA-NeMo/NeMo#2730] [Q&A] [answered]
Hi,
I want to export the model from `ckpt` for testing during model training.
Is there any script to export `model.nemo` from `ckpt`?

Many thank!

**Comment by VahidooX** [comment, marked as answer]
You need to load the model with method load_from_checkpoint and then save it as nemo file. For instance if it is an ASR model, you may do it like this:

```
    model = nemo.collections.asr.models.EncDecCTCModelBPE.load_from_checkpoint(model_ckpt)
    model.save_to(output_path)

```

You may also set the following in your config, so checkoints are stored as nemo file instead of PTL checkpoints:

`exp_manager.checkpoint_callback_params.always_save_nemo = true`

**Reply by Arkymeng** [reply]
hello, why there occurs  **' ValueError: `cfg` must have `tokenizer` config to create a tokenizer !'** when I load model from  ckpt?

**Comment by Akshaya-Devadiga** [comment]
code: 
import nemo.collections.nlp as nemo_nlp
model = nemo.collections.nlp.models.distilbert-base-uncased.load_from_checkpoint('pre_trained_bert/checkpoints/SequenceClassifier-STEP-7210.pt')

ERROR:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-74-eec83ed8f27a> in <module>
----> 1 model = nemo.collections.nlp.models.distilbert-base-uncased.load_from_checkpoint('pre_trained_bert/checkpoints/SequenceClassifier-STEP-7210.pt')
      2 #model.save_to(output_path)

AttributeError: module 'nemo.collections.nlp' has no attribute 'models'

@VahidooX could u plz help me solve this error

**Reply by VahidooX** [reply]
You need to use the name of the model class instead of distilbert-base-uncased. Would you please try the following:

`model = nemo.collections.nlp.models.TokenClassificationModel.load_from_checkpoint('pre_trained_bert/checkpoints/SequenceClassifier-STEP-7210.pt')`
